src/pipeline.rs

```diff
@@ -382,3 +430,98 @@ pub(crate) fn prepare_view_targets(
         }
     }
 }
+
+#[derive(Resource, Clone)]
+pub struct PointCloudPlaybackControl {
+    pub playing: bool,
+    pub speed: f32,
+}
+impl Default for PointCloudPlaybackControl {
+    fn default() -> Self {
+        Self {
+            playing: true,
+            speed: 5.0,
+        }
+    }
+}
+
+impl ExtractResource for PointCloudPlaybackControl {
+    type Source = Self;
+
+    /// Defines how the resource is transferred into the "render world".
+    fn extract_resource(source: &Self::Source) -> Self {
+        source.clone()
+    }
+}
+
+pub fn prepare_animated_assets(
+    queue: Res<RenderQueue>,
+    mut assets: ResMut<RenderAssets<PointCloudAsset>>,
+    playback: ResMut<PointCloudPlaybackControl>,
+    time: Res<Time>,
+) {
+    if !playback.playing {
+        return;
+    }
+    for (_handle, asset) in assets.iter_mut() {
```

Comment:
IMO this goes against the asset abstraction. Instead of mutating the assets directly, you should spawn entities with a `Handle<PointCloudAsset>` component and iterate over a query of those. Then you can have a separate component for the playback fields, probably merged with `PointCloudPlaybackControl`.

Reply:
The problem is that there isn't a good place to store the animation buffer. Like you said optimally we'd store that in the render world as a component, alongside the playback control component. Problem is bevy clears all render world components every frame, so now we're required to put it into a resource. This quickly gets ugly.

The downside with this approach is that for each instance of the same opd file, they must be animated together. But we're not required to animate multiple entities for now, so putting that inside the render asset becomes the nature thing to do.

We can revisit this when it becomes a problem.

Reply:
OK. Can you make an issue in this repo for this change? I think it's important to follow the bevy abstractions, it's just unfortunate that bevy makes it hard to use render world entities. The alternative is to store a `Res<HashMap<Entity, AnimationBuffer>>` in the render world.